src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -158,20 +175,66 @@ protected override async Task OnClientMessageAsync(ConnectionDataMessage connect
             }
         }
 
+        private async Task SkipHandshakeResponse(ClientConnectionContext connection)
+        {
+            try
+            {
+                while (true)
+                {
+                    var result = await connection.Application.Input.ReadAsync();
+                    if (result.IsCanceled)
+                    {
+                        return;
+                    }
+
+                    var buffer = result.Buffer;
+                    if (buffer.IsEmpty)
+                    {
+                        continue;
+                    }
+
+                    if (SignalRProtocol.HandshakeProtocol.TryParseResponseMessage(ref buffer, out var message))

```

Comment:
>TryParseResponseMessage [](start = 58, length = 23)

shall we log something when the first response is not handshake and just stop the connection?

Reply:
So the `OnConnected` hub event will be triggered on this "migrated to" app server, right? will `OnDisconnected` hub event on the "migrated from" app server?

---
In reply to: [351961944](https://github.com/Azure/azure-signalr/pull/739#discussion_r351961944) [](ancestors = 351961944)